core/embed/rust/src/ui/model_mercury/flow/confirm_action.rs

```diff
@@ -240,53 +306,69 @@ fn create_flow(
 ) {
     let prompt_screen = prompt_screen.or_else(|| hold.then_some(title));
     let prompt_pages: usize = prompt_screen.is_some().into();
-
-    let (flow, page): (Result<SwipeFlow, Error>, &dyn FlowController) = if prompt_screen.is_some() {
-        (SwipeFlow::new(&ConfirmAction::Intro), &ConfirmAction::Intro)
-    } else {
-        (
-            SwipeFlow::new(&ConfirmActionSimple::Intro),
-            &ConfirmActionSimple::Intro,
-        )
+    let initial_page: &dyn FlowController = match (extra, prompt_screen.is_some()) {
+        (ConfirmActionExtra::Menu { .. }, false) => &ConfirmActionWithMenu::Action,
+        (ConfirmActionExtra::Menu { .. }, true) => &ConfirmActionWithMenuAndConfirmation::Action,
+        _ => &ConfirmAction::Action,
     };
 
-    (prompt_screen, prompt_pages, flow, page)
+    (
+        prompt_screen,
+        prompt_pages,
+        SwipeFlow::new(initial_page),
+        initial_page,
+    )
 }
 
 fn create_menu(
     flow: SwipeFlow,
-    menu: ConfirmActionMenu,
+    extra: ConfirmActionExtra,
     prompt_screen: Option<TString<'static>>,
 ) -> Result<SwipeFlow, Error> {
-    let mut menu_choices = VerticalMenu::empty().danger(
-        theme::ICON_CANCEL,
-        menu.verb_cancel.unwrap_or(TR::buttons__cancel.into()),
-    );
-
-    if menu.info {
-        menu_choices = menu_choices.item(
-            theme::ICON_CHEVRON_RIGHT,
-            menu.verb_info
-                .unwrap_or(TR::words__title_information.into()),
+    if let ConfirmActionExtra::Menu {
+        verb_cancel,
+        has_info,
+        verb_info,
+    } = extra
+    {
+        // NB: The Cancel menu item is always the first,
+        // because of the MENU_ITEM_CANCEL = 0.
+        // If we want the cancel item to be somewhere else,
+        // we would need to account for that and we could not use a constant.
+        let mut menu_choices = VerticalMenu::empty().danger(
+            theme::ICON_CANCEL,
+            verb_cancel.unwrap_or(TR::buttons__cancel.into()),
```

Comment:
Not sure what is the benefit. It would just move the check / unwrap at a higher level, most probably in `ConfirmBlobParams` - because that one *can* have it missing. Let's say I also set it to `TR::buttons__cancel ` in `ConfirmBlobParams`, I would still need to make the check in `new_confirm_action`. Unless I make it non-optional there too and set a default in uPy... Either way, the default would have to be somewhere, no?

Reply:
OK: fc0bb630620df3c133a12e9bd437fdc6199d62d6

I think I understand your point, and it can be seen like this too, indeed.

I was seeing it from the other angle - as much as possible instantiate these "string bearing structs" like `ConfirmBlobParams` and `ConfirmActionMenuStrings` (maybe I should rename it to `ConfirmActionMenuParams` while at it...) with no parameters, have everything default to None, and do `with_XXX` only when "XXX" needs to be changed to something other than the default. Otherwise the struct would carry the None all the way to the very end, where the default would be used. I think @obrusvit 's recent changes also went in this direction - dropping as many parameters from `ConfirmBlobParams::new` and moving them to `.with_XXX`.

Of course, the two ways are not mutually exclusive, which is what I tried to do here - not sure whether you guys like it - but basically `ConfirmBlobParams::new` still means you want to go with defaults, but the default `verb_cancel` is set to `TR::buttons__cancel.into()` in the ctor rather than to `None` - so this "params bearing struct" carries the concrete default around. Maybe if this is what you meant, then I should just replace the rest of the strings in `ConfirmBlobParams` in the same way... I think I like it...

The disadvantage is of course that multiple instantiations of `ConfirmBlobParams` around multiple API functions have to `unwrap_or` *if* they want to do a `.with_verb_cancel` (in order to let the uPy code customize the verb. OOOOR, I could make `with_verb_cancel` accept an Option and have *it* set the default in case it received `None` - but I don't know why, that looks less appealing to me (even though the opposite leads to duplication of the default value).

Reply:
> Maybe if this is what you meant, then I should just replace the rest of the strings in `ConfirmBlobParams` in the same way... I think I like it...

yes, this is what I have in mind

> multiple instantiations of ConfirmBlobParams around multiple API functions have to unwrap_or if they want to do a .with_verb_cancel (in order to let the uPy code customize the verb. OOOOR, I could make with_verb_cancel accept an Option and have it set the default in case it received None - but I don't know why, that looks less appealing to me (even though the opposite leads to duplication of the default value).

wondering about this .. would it make sense to construct `ConfirmBlobParams` and friends by passing the micropython options -- gated by `#[feature(micropython)]` of course?
wdyt @obrusvit